app/services/permissions_checker.rb

```diff
@@ -1,17 +1,25 @@
 # frozen_string_literal: true
 
 class PermissionsChecker
-  def initialize(current_user:, permission_names:, user_id:, friendly_id:, record_id:)
+  def initialize(current_user:, permission_names:, user_id:, friendly_id:, record_id:, current_provider:)
     @current_user = current_user
     @permission_names = permission_names
     @user_id = user_id
     @friendly_id = friendly_id
     @record_id = record_id
+    @current_provider = current_provider
   end
 
   def call
+    # check to see if current_user has SuperAdmin role
     return true if @current_user.role == Role.find_by(name: 'SuperAdmin', provider: 'bn')
 
+    # check to see if current user is trying to access own provider info
+    return false if @current_user.provider != @current_provider
+
+    # check to see if current_user is trying to access a user from another provider
+    return false if @user_id.present? && (@current_user.provider != User.find(@user_id.to_s).provider)
```

Comment:
Would putting those two checks in a separated method solve the Rubocop complexity error?

Reply:
Maybe extract line 15 as well

Reply:
We could, but it would just lower the number of possible outcomes in the method. I feel we should either leave them all as is, or take lines 14-27 and put them into a function

Reply:
Everytime I had those Rubocop error, I found a way to rewrite the logic in a simpler/more satisfying way.
Could be a bit harder to do here as this is one of the most complex service we have.
I would definetly try to avoid lowering the Rubocop constraint.

Reply:
Don't we want to further check the rest of resources like rooms, recordings?
Like with now we're checking if a user is trying to manage a user from another organazation we want to do that also for the rest of resources.
Maybe extract that into a private method like a `check_provider` or something more relevant.
That should help reduce the complexity and improve maintainability and readibility.
That may be a bit challenging to make the method communicate with the rest of logic (if returns false that won't fail the check as is now).
Remember to always start first with `return false if @current_user.provider != @current_provider`.